components/webapp-mgt/org.wso2.carbon.webapp.mgt/src/main/java/org/wso2/carbon/webapp/mgt/filter/AuthorizationHeaderFilter.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.webapp.mgt.filter;
+
+import org.apache.axis2.transport.http.HTTPConstants;
+import org.apache.commons.lang.StringUtils;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletContext;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.annotation.WebFilter;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Enumeration;
+
+/**
+ * Servlet Filter implementation class for authorization header validation. This handles the NPE thrown by CXF
+ * transport exposing the stack-trace and server information to the end user for authorization headers with invalid
+ * encoded json data by sending a proper error response back to the user. If there is more than single space between
+ * auth_type and json data this will modify the header to be in correct format and pass the request along the filter
+ * chain.
+ *
+ * @since 4.7.6
+ */
+@WebFilter("/AuthorizationHeaderFilter")
+public class AuthorizationHeaderFilter implements Filter {
+
+    private static final String AUTH_TYPE_BASIC = "Basic";
+    private static final int AUTH_TYPE_BASIC_LENGTH = 5; //length of "Basic" String
+    private static final int MINIMUM_CREDENTIAL_SIZE = 4;
+
+    private ServletContext context;
+
+    public void init(FilterConfig fConfig) throws ServletException {
+
+        this.context = fConfig.getServletContext();
+        this.context.log("AuthorizationHeaderFilter initialized");
+    }
+
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
+            throws IOException, ServletException {
+
+        final HttpServletRequest req = (HttpServletRequest) request;
+        final String authHeader = req.getHeader(HTTPConstants.HEADER_AUTHORIZATION);
+        if (StringUtils.isEmpty(authHeader)) {
+            chain.doFilter(request, response);
+            return;
+        }
+
+        String authType = null;
```

Comment:
https://github.com/wso2/carbon-deployment/pull/273/files#diff-205dffde3863e7eeeef78c1788679b98R78 gives an error if authType is not initialized

Reply:
Can you explain ?

Reply:
Since authType is a local variable it doesn't have a default value, it should be assigned a default value before it can be used to avoid compilation errors.

Reply:
The default value of the String is anyway NULL, you do not have to assign it explicitly.

Reply:
The default value of the String is anyway NULL, you do not have to assign it explicitly.

Reply:
What you are saying is applicable to instance variables, since this is a local variable they are in undefined states if not explicitly defined. Otherwise this gives a compilation error.

Reply:
Yes, default values are assigned for the variables of 8 primitive data types and Strings, except for local variables. Please refer to oracle documentation [here](https://docs.oracle.com/javase/tutorial/java/nutsandbolts/variablesummary.html)

> The compiler will assign a reasonable default value for fields of the above types; **for local variables, a default value is never assigned**

Reply:
Got it. Agreed!